tests/cpp/data/test_simple_dmatrix.cc

```diff
@@ -126,3 +126,32 @@ TEST(SimpleDMatrix, EmptyRow) {
   CHECK_EQ(dmat.Info().num_row_, 2);
   CHECK_EQ(dmat.Info().num_col_, 2);
```

Comment:
you may also consider changes that went into [test_adapter.cc](https://github.com/dmlc/xgboost/pull/5097/files#diff-75016bcb82696e4ac5cdb4a33ba99a8a)